src/App.tsx

```diff
@@ -1,61 +1,41 @@
+import { useState } from 'react';
+
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { PreparedTodo } from './types';
+
+import { TodoAddForm } from './components/TodoAddForm';
+import { TodoList } from './components/TodoList';
+
+const preparedTodoTasks: PreparedTodo[] = todosFromServer.map(todo => {
+  const user = usersFromServer.find(({ id }) => todo.userId === id) || null;
+
+  return {
+    ...todo,
+    user,
+  };
+});
 
 export const App = () => {
+  const [todos, setTodos] = useState(preparedTodoTasks);
+
+  const addTask = (newTask: PreparedTodo): void => {
+    setTodos((prevState) => [...prevState, newTask]);
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
-
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
-
-          <span className="error">Please choose a user</span>
-        </div>
-
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
-
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">
-            delectus aut autem
-          </h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
-
-          <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-            Patricia Lebsack
-          </a>
-        </article>
-      </section>
+      <TodoAddForm
+        users={usersFromServer}
+        todos={todos}
+        addTask={addTask}
```

Comment:
```suggestion
        onAddTodo={handleAddTodo}
```

Reply:
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364